single-node-refactor/src/Solvers/SGH_solver/include/contact.h

```diff
@@ -0,0 +1,148 @@
+#pragma once
+
+#include "matar.h"
+#include "mesh.h"
+#include "_debug_tools.h"  // Remove this file entirely once finished
+
+using namespace mtr;
+
+struct contact_patch_t
+{
+    size_t gid;  // global patch id
+    CArrayKokkos<size_t> nodes_gid;  // global node ids
+
+    /*
+     * If the position of a point is denoted by "p" and "p" is a vector p = (px, py, pz), then the following arrays
+     * are structured as follows:
+     *
+     * ⎡p_{0x}  p_{1x}  p_{2x} ... p_{nx}⎤
+     * ⎢                                 ⎥
+     * ⎢p_{0y}  p_{1y}  p_{2y} ... p_{ny}⎥
+     * ⎢                                 ⎥
+     * ⎣p_{0z}  p_{1z}  p_{2z} ... p_{nz}⎦
+     *
+     * For a standard linear hex, this matrix is a 3x4. vel_points is structured the same way.
+     */
+    CArrayKokkos<double> points;  // coordinate points of patch nodes
+    CArrayKokkos<double> vel_points;  // velocity of patch nodes
+    CArrayKokkos<double> internal_force;  // any force that is not due to contact (corner force)
+
+    // Iso-parametric coordinates of the patch nodes (1D array of size mesh.num_nodes_in_patch)
+    // For a standard linear hex, xi = [-1.0, 1.0, 1.0, -1.0], eta = [-1.0, -1.0, 1.0, 1.0]
+    // For now, these are the same for all patch objects, but should they be different, then remove static and look to
+    // contact_patches_t::initialize for how to set these values
+    CArrayKokkos<double> xi;  // xi coordinates
+    CArrayKokkos<double> eta;  // eta coordinates
+    static size_t num_nodes_in_patch;  // number of nodes in the patch (or surface)
+
+    /*
+     * Updates the points and vel_points arrays. This is called at the beginning of each time step in the
+     * contact_patches_t::sort() method.
+     *
+     * @param nodes: node object that contains coordinates and velocities of all nodes
+     */
+    KOKKOS_FUNCTION
+    void update_nodes(const mesh_t &mesh, const node_t &nodes, const corner_t &corner);
+
+    /*
+     * The capture box is used to determine which buckets penetrate the surface/patch. The nodes in the intersecting
+     * buckets are considered for potential contact. The capture box is constructed from the maximum absolute value
+     * of velocity and acceleration by considering the position at time dt, which is equal to
+     *
+     * position + velocity_max*dt + 0.5*acceleration_max*dt^2 and
+     * position - velocity_max*dt - 0.5*acceleration_max*dt^2
+     *
+     * The maximum and minimum components of the capture box are recorded and will be used in
+     * contact_patches_t::find_nodes.
+     *
+     * @param vx_max: absolute maximum x velocity across all nodes in the patch
+     * @param vy_max: ||        ||        ||        ||        ||        ||
```

Comment:
With doxygen, you will need to write these out. If not, the formatting on the generated webpage wont necessarily capture the || to be inline with the above sentence. It will also change if we use sphinx+doxygen, which I think we plan to in the mid to distant future lol.